onnx/onnx.in.proto

```diff
@@ -370,14 +374,19 @@ message TensorProto {
   repeated uint64 uint64_data = 11 [packed = true];
 }
 
-// Defines a tensor shape. A dimension can be either an integer value
-// or a symbolic variable. A symbolic variable represents an unknown
-// dimension.
+// Defines a tensor shape. A dimension can be
+//   - an integer value
+//   - a symbolic variable. All symbolic dimensions in the graph with
+//     the same name are constrained to take on the same value.
+//   - unspecified. Unspecified dimensions have no constraints on
+//     them. They are equivalent to symbolic dimensions with unique,
+//     autogenerated names.
```

Comment:
I'd keep current design to use unique symbolic name in this case as implied here, other than having another flag (enum).